### DISCUSSION TITLE
Accessing ROCKET attributes

### DISCUSSION THREAD

**Discussion by chrico-bu-uab** [alan-turing-institute/sktime#5849] [Ideas]
Having run models on ROCKET-transformed time series data, it would be nice to know the specifics of what went into each feature, for instance if top features have similar weights, length, biases, dilation, and/or padding.

I noticed that an older package pyts has a ROCKET class. There the weights, etc. are attributes accessible to the user. Could we do this here for our various ROCKET classes?

Update: I just learned about the `kernels` attribute, which contains all the components, though it would still be nice to have individual attributes for the individual components. Currently it is kind of a pain to extract this information.

**Comment by fkiraly** [comment]
Would you like to open a feature request issue on this?
Make sure to ping the current maintainer and authors of the algorithm, you can get them by `get_tags()["maintainers"]` on an instance (or `get_class_tags` for the class). In this case, @angus924.

Generally, `get_fitted_params` catches all attributes ending in underscore, in line with `sklearn` convention. In this case, `kernels` is neither documented nor visible (because it does not end in undescore).
If you would like to help out, you could make a mini-PR that copies the attribute to `self.kernels_` (without removing the original), and adds it to the `Attributes` section in the docstring (with an explanation of the format), while I look into `pyts`.

I've also opened an issue on interfacing `pyts` in general: https://github.com/sktime/sktime/issues/5850
(had it on the radar for a while, but did not record the idea)

**Comment by fkiraly** [comment]
`pyts`/`ROCKET` interface added here: https://github.com/sktime/sktime/pull/5851

Perhaps that solves your requirement, at least it does not hurt to try multiple things at the same time.
